webapp/src/marked.tsx

```diff
@@ -350,6 +352,105 @@ export class MarkedContent extends data.Component<MarkedContentProps, MarkedCont
             })
     }
 
+    // Renders collapsable hints starting with "~hint" and ending with "hint~".
+    // Example input:
+    // -------------------------------
+    // ~hint This is the hint summary.
+    //
+    //    This is the hint content
+    //
+    // hint~
+    // -------------------------------
+    //
+    // Example output:
+    // -------------------------------
+    // <details><summary>This is the hint summary</summary>
+    //   This is the hint content
+    // </details>
+    // -------------------------------
+    private renderAccordianHints(content: HTMLElement) {
+        const hintBeginRegex = /^\s*~hint\s*(.+)/i;
+        const hintEndRegex = /^\s*hint~.*/i;
+
+        // Processing each hint modifies the node order and count of 'content.childnodes'.
+        // Keep reprocessing content.childnodes until all hints have been processed.
+        let hintFound: boolean = false;
+
+        do {
```

Comment:
This is bit hard to read for me. Can we do this loop in two steps?

1. First loop finds all the hints
2. Second loop acts on the hints it found in the first loop.

Are we supporting nested hints here?

Reply:
We are not supporting nested hints. I have refactored the loops into two steps.